config_parser.cc

```diff
@@ -199,18 +201,21 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
           new_config);
       config_stack.push(new_config);
     } else if (token_type == TOKEN_TYPE_END_BLOCK) {
+      block_depth--;
       if (last_token_type != TOKEN_TYPE_STATEMENT_END) {
         // Error.
         break;
       }
       config_stack.pop();
     } else if (token_type == TOKEN_TYPE_EOF) {
       if (last_token_type != TOKEN_TYPE_STATEMENT_END &&
-          last_token_type != TOKEN_TYPE_END_BLOCK) {
+          last_token_type != TOKEN_TYPE_END_BLOCK &&
+	  last_token_type != TOKEN_TYPE_START) {
         // Error.
         break;
       }
-      return true;
+      if (block_depth == 0)
+          return true;
     } else {
```

Comment:
The varied indentations in the lines 212-218 above make the code slightly harder to read; I recommend standardizing them.